src/test/sql/query_plan_cache_test.cpp

```diff
@@ -143,4 +143,24 @@ TEST_F(QueryPlanCacheTest, AutomaticQueryOperatorCacheLRUK2) {
   EXPECT_EQ(5u, _query_plan_cache_hits);
 }
 
+// Check access to PQP cache. When set, check the underlying cache implementation, and verify that it is a GDFS cache
+// that supports retrieving the cache frequency count.
+TEST_F(QueryPlanCacheTest, CachedPQPFrequencyCount) {
+  // Create pipeline and pass pqp cache. Verify that this does not change default_pqp_cache.
+  auto sql_pipeline = SQLPipelineBuilder{Q1}.with_pqp_cache(cache).create_pipeline_statement();
+  EXPECT_EQ(SQLPipelineBuilder::default_pqp_cache, nullptr);
+
+  // Setting default_pqp_cache and verify it's set.
+  SQLPipelineBuilder::default_pqp_cache = cache;
```

Comment:
This looks like it might break test isolation? Also, should Hyrise::reset reset the default caches?

Reply:
The clean up executes:
```
~BaseTestWithParam() {
    Hyrise::reset();
    SQLPipelineBuilder::default_pqp_cache = nullptr;
    SQLPipelineBuilder::default_lqp_cache = nullptr;
  }
```

Thus, I think it's safe here. Now that you ask, I'd vote for `Hyrise::reset()` resetting the default caches. But I am not sure what the motivation was not to do it.

Reply:
I'll address it with the latest issue on BaseTest?

Reply:
If you agree, we can resolve and proceed.

Reply:
Looks like the other thing will be part of #1986, so maybe do this one here?